elements/pl-code/pl-code.py

```diff
@@ -202,19 +232,18 @@ def render(element_html, data):
         #
         # which technically starts with a newline, but we probably
         # don't want a blank line at the start of the code block.
-        code = pl.inner_html(element)
-        if len(code) > 1 and code[0] == "\r" and code[1] == "\n":
-            code = code[2:]
-        elif len(code) > 0 and (code[0] == "\n" or code[0] == "\r"):
-            code = code[1:]
+        code = pl.inner_html(element).removeprefix("\r").removeprefix("\n")  # type: ignore
```

Comment:
Consider annotating `pl.inner_html` with a return type to remove the need for `# type: ignore` here; same with above.

Reply:
That's not the issue, the `removeprefix` and `removesuffix` functions aren't getting recognized by pyright for some reason.

Reply:
The issue is not `pl.inner_html`, it's `removeprefix`, which is a feature introduced to `str` in Python 3.9 that apparently is not yet supported by pyright for some reason.

Reply:
I think i know what's up actually, give me a sec to push.

Reply:
We pinned pyright to 3.8, so it makes sense that 3.9 features wouldn't be available: https://github.com/PrairieLearn/PrairieLearn/blob/c50f199cb622b2721bd64cb7d3b7ac55aeadb7d5/pyproject.toml#L78. Let's bump that to 3.9? That's the oldest version currently in production.

Reply:
@nwalters512 read my mind https://github.com/PrairieLearn/PrairieLearn/pull/7185/commits/93647d71907ba271d27938dffd9685d473b789c2

Going to go ahead and change to the simpler type annotations as well. I thought we were on 3.10 though?

Reply:
Not everywhere; I still need to find the bandwidth to get us updated in Canada.